tmt/steps/provision/testcloud.py

```diff
@@ -479,10 +479,15 @@ def start(self):
                 f"Failed to prepare the image. Check the '{TESTCLOUD_IMAGES}' "
                 f"directory permissions.", original=error)
 
+        # Prepare hostname (get rid of possible non-ascii characters)
```

Comment:
@psss could you describe a bit the logic or add some examples here, I wanted to suggest other solutions, but this comment does not fully help me understand what you were trying to achieve and why it has to be like this

Reply:
The `name` key can potentially contain spaces and basically any characters:

```yaml
# Request two guests
provision:
  - name: server
    how: virtual
  - name: What's possible to put here? Basicaly anything!
    how: virtual
```

So here, we just replace all non-ascii stuff with a dash.

Reply:
I wouldn't call it "non-ascii" - `!` or `?` are defined by ASCII, but we don't need them either. On the other hand, `-` and digits are perfectly fine, and `name: foo01` should be supported, because `foo01` is absolutely valid hostname.

I don't understand why three slightly different regular expressions are used, a comment explaining that would be nice, although I would propose a slightly different expression eventually:

```python
unwanted_pattern = re.compile(r'[^a-zA-Z0-9\-]')

for s in [
    "What's possible to put here? Basicaly anything!",
    "server",
    "foo01",
    "bar_7-baz11"
]:
    print(f'"{s}" => "{unwanted_pattern.sub("-", s)}"')
```

```
"What's possible to put here? Basicaly anything!" => "What-s-possible-to-put-here--Basicaly-anything-"
"server" => "server"
"foo01" => "foo01"
"bar_7-baz11" => "bar-7-baz11"
```

* uppercase letters, digits and `-`are allowed to appear in hostnames
* expression focuses on a single character at a time, no matter where in the string it lies

Reply:
Thanks for providing the better regular expression. The three substitutions were present to get rid of the unwated characters from the beginning and end of the string. However, this can be easily done by applying a `strip("-")` on the substituted string. Updated.